src/lighteval/tasks/registry.py

```diff
@@ -230,84 +272,75 @@ def task_registry(self) -> dict[str, LightevalTaskConfig]:
-                format: "task_definition", where it can be
-                containing a list of tasks.
-                where task_definition can be:
-                - path to a file containing a list of tasks (one per line)
-                - task group defined in TASKS_GROUPS dict in custom tasks file
-                - task name with few shot in format "suite|task|few_shot|truncate_few_shots"
-                - task superset in format "suite|task_superset|few_shot|truncate_few_shots" (superset will run all tasks with format "suite|task_superset:{subset}|few_shot|truncate_few_shots")
-
-
-        Returns:
-            tuple[list[str], dict[str, list[tuple[int, bool]]]]: A tuple containing:
-                - A sorted list of unique task names in the format "suite|task".
-                - A dictionary mapping each task name to a list of tuples representing the few_shot and truncate_few_shots values.
+        Updates each config depending on the input tasks (we replace all provided params, like few shot number, sampling params, etc)
         """
-        task_to_params = collections.defaultdict(list)
-
-        # We can provide a path to a file with a list of tasks or a string of comma-separated tasks
-        if os.path.exists(tasks):
-            with open(tasks, "r") as f:
-                tasks_list = [line.strip() for line in f if line.strip() and not line.startswith("#")]
-        else:
-            tasks_list = tasks.split(",")
+        task_to_configs = collections.defaultdict(list)
 
-        # At this point the strings are either task name/superset name or group names
-        # Here we deal with group names and map them to corresponding tasks
-        expanded_tasks_list: list[str] = []
-        for maybe_task_group in tasks_list:
-            # We either expand the group (in case it's a group name), or we keep it as is (in case it's a task name or superset name)
-            expanded_tasks = self.task_groups_dict.get(maybe_task_group, [maybe_task_group])
-            if len(expanded_tasks) > 1:
-                logger.info(f"Expanding task group {maybe_task_group} to {expanded_tasks}")
-            expanded_tasks_list.extend(expanded_tasks)
-
-        for task in expanded_tasks_list:
+        # We map all tasks to their parameters
+        for task in self.tasks_list:
             metric_params_dict = {}
             try:
-                suite_name, task_name, few_shot, truncate_few_shots = tuple(task.split("|"))
+                if task.count("|") == 3:
+                    logger.warning(
+                        "Deprecation warning: You provided 4 arguments in your task name, but we no longer support the `truncate_fewshot` option."
+                    )
+                    logger.warning(
+                        "We will ignore the parameter for now, but it will fail in a couple of versions, so you should change your task name to `suite|task|num_fewshot`."
+                    )
```

Comment:
```suggestion
                    logger.warning(
                        "Deprecation warning: You provided 4 arguments in your task name, but we no longer support the `truncate_fewshot` option. We will ignore the parameter for now, but it will fail in a couple of versions, so you should change your task name to `suite|task|num_fewshot`."
                    )